FontEditor+LibGUI: Highlight modified glyphs

AtkinsSJ opened the pull request:
This makes modifications in FontEditor more visible, both so you know
what you've changed, and for taking a handy "here's what's changed"
screenshot for a font PR. :^)

The background color for new glyphs is green, modified glyphs is blue,
and deleted glyphs is red. The changes persist until you load a new
font file, so you can continue saving your work as you go and still be
able to take a convenient screenshot at the end.

----

Now with dark-mode colours, proper undo support, and an option to turn the highlights on and off! :^)

![image](https://user-images.githubusercontent.com/222642/182388274-9f6d3c75-9ef4-4c9b-b53c-2425ef67a32f.png)

![image](https://user-images.githubusercontent.com/222642/182388354-b3920338-fdd9-41ce-b92d-12aa336b49fc.png)

djwisdom commented:
More love to FontEditor. ❤️

TobyAsE reviewed (approved):
Lgtm, very nice addition!

Some sort of documentation would be nice as to not guess what the colours mean. Tooltip maybe, or a legend somewhere... But that's :ideaguying:

Xexxa reviewed (changes requested):
I love this addition. I would love it even more if there was an on/off option.

AtkinsSJ commented:
Thanks everyone! I wasn't initially sure if this was a desired feature. I've improved it a bit:
- Tweaked the colours because I wasn't happy with them.
- Added darker versions of the colours when the theme is dark. (A proper solution might be to add color-roles for them, but it felt excessive for one application. If we end up with other apps wanting "created/modified/deleted" colours then we can move it into the theme then.)
- Added undo support.
- Add a menu option to show or hide the highlights.

New screenshots are in the first post. :^)

AtkinsSJ commented:
Fixed a merge conflict.
